### DISCUSSION TITLE
Is nano-jsx right for me?

### DISCUSSION THREAD

**Discussion by erfanium** [nanojsx/nano#126] [Help / Questions] [answered]
I want to generate some html files in server-side. I used nano-jsx as an alternative to ejs templates.
My code is not ever required to be executed in browser. also I've never need to use react hooks.

I found this module the tiniest module to do it. but this module also includes browser/DOM support that i don't want.
Wonder if is there any alternatives for me?

Thanks

**Comment by yandeu** [comment, marked as answer]
Well, if you just want to use it on the server-side, that perfectly fine. There is no disadvantage of doing so.

**Comment by dcflow** [comment]
Hi @yandeu ,

Can you make the project work without TypeScript and the "tsconfig"? What is the most minimal config 
to set it up?

I have a simple node/nodemon project, and I want to use nano-jsx because for me also seems to be the best option
out there.

Thank you for your help!

**Reply by yandeu** [reply]
I am not sure if you can do it without a `tsconfig`. Maybe using babel and webpack?
Anyways I highly recommend using the TypeScript compiler, which does not require you writing TypeScript code.
I just made an minimal example for Nodejs using JavaScript:
https://github.com/nanojsx/minimal-setup

Hope this helps :)

---

Update:
And here, an minimal example for the browser:
https://github.com/nanojsx/minimal-setup-browser

**Reply by dcflow** [reply]
Awesome, this is a great starting point to build on top of it  🙏  Thank you very much for your quick response, 
and solution :)
